consensus/src/checkpoint.rs

```diff
@@ -3,57 +3,18 @@
 
 use crate::common::{Chain, ProtocolStateChain};
 use crate::constants::SLOTS_PER_EPOCH;
-use mina_crypto::base58::Base58Encodable;
-use mina_crypto::hash::{EpochSeed, StateHash};
-use mina_rs_base::protocol_state::ProtocolState;
-
-#[derive(Debug)]
-pub enum ConsensusErrTyp {
-    ConsensusInitFail,
-}
-pub fn init_checkpoints(genesis: &mut ProtocolState) -> Result<(), ConsensusErrTyp> {
-    genesis.body.consensus_state.staking_epoch_data.seed = EpochSeed::default();
-    genesis
-        .body
-        .consensus_state
-        .staking_epoch_data
-        .start_checkpoint = StateHash::default();
-    genesis
-        .body
-        .consensus_state
-        .staking_epoch_data
-        .lock_checkpoint = StateHash::default();
-    genesis
-        .body
-        .consensus_state
-        .staking_epoch_data
-        .epoch_length
-        .0 = 1;
-    genesis.body.consensus_state.next_epoch_data.seed =
-        EpochSeed::from_base58("2vaRh7FQ5wSzmpFReF9gcRKjv48CcJvHs25aqb3SSZiPgHQBy5Dt")
-            .map_err(|_| ConsensusErrTyp::ConsensusInitFail)?;
-    genesis
-        .body
-        .consensus_state
-        .next_epoch_data
-        .start_checkpoint = StateHash::default();
-    genesis.body.consensus_state.next_epoch_data.lock_checkpoint =
-        StateHash::from_base58("3NLoKn22eMnyQ7rxh5pxB6vBA3XhSAhhrf7akdqS6HbAKD14Dh1d")
-            .map_err(|_| ConsensusErrTyp::ConsensusInitFail)?;
-    genesis.body.consensus_state.next_epoch_data.epoch_length.0 = 2;
-    Ok(())
-}
```

Comment:
Uncertain if we can totally ignore this lifecycle for checkpointing?

Reply:
I could not follow the question, could you rephrase?

Reply:
I'm wondering if in Blockchain Mina protocol checkpoint has a life cycle as [Life Cycle of a Servlet](https://www.geeksforgeeks.org/life-cycle-of-a-servlet/) has to init(), resume(),  and destroy()? Since we have init_checkpoints() to init its states. If yes we will need to handle the life cycle with care. Probably will not just throw it into the test module.

Reply:
No they are not related to servlet lifecycle methods. As far as i see it, the init_checkpoint function was just a helper function created to make writing test cases easier by helping stub the staking epoch and next epoch fields.